netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -290,16 +308,14 @@ static ProtocolNegotiator createProtocolNegotiator(
 
     private boolean closed;
 
-    private NettyTransportFactory(Class<? extends Channel> channelType,
-                                  NegotiationType negotiationType,
-                                  ProtocolNegotiator protocolNegotiator,
-                                  SslContext sslContext,
-                                  EventLoopGroup group,
-                                  int flowControlWindow,
-                                  int maxMessageSize,
-                                  int maxHeaderListSize) {
+    private NettyTransportFactory(
+        Class<? extends Channel> channelType, Map<ChannelOption<?>, Object> channelOptions,
+        NegotiationType negotiationType, ProtocolNegotiator protocolNegotiator,
+        SslContext sslContext, EventLoopGroup group, int flowControlWindow, int maxMessageSize,
+        int maxHeaderListSize) {
       this.channelType = channelType;
       this.negotiationType = negotiationType;
+      this.channelOptions = channelOptions;
```

Comment:
Only `NettyChannelBuilder#withOption(ChannelOption<T> option, T value)` can mutate it. I think that is acceptable, after all, calling `withOption()` after `build()` to mutate it is anti-pattern. 
Declaring the field as` Map<ChannelOption<?>, ?>` can make it not modifiable inside the current class.